lib/system.nim

```diff
@@ -1127,12 +1127,9 @@ const
     ## is the value that should be passed to `quit <#quit,int>`_ to indicate
     ## failure.
 
-when defined(js) and defined(nodejs) and not defined(nimscript):
-  var programResult* {.importc: "process.exitCode".}: int
-  programResult = 0
```

Comment:
Is this the bugfix? Why didn't this work?

Reply:
it's one of the bugfixes; before PR, `programResult = 1` was a noop

Reply:
Why did it not work with the importc way though?

Reply:
probably because exitCode is a special variable, eg:
```nim
  {.emit:"""
  function bar0(){
    console.log("exitCode", process.exitCode);
    process.exitCode = 1;
    console.log("exitCode", process.exitCode);
  }
  """.}

  proc bar0(){.importc:"bar0".}

  proc main()=
    bar0()
  main()
```

this prints:
```
exitCode undefined
exitCode 1
```
instead of:
```
exitCode 0
exitCode 1
```